src/main/target/MATEKF722/target.c

```diff
@@ -30,7 +30,7 @@
 const timerHardware_t timerHardware[USABLE_TIMER_CHANNEL_COUNT] = {
     DEF_TIM(TIM9, CH2, PA3, TIM_USE_PPM, 0, 0),      // PPM
 
-    DEF_TIM(TIM8, CH1, PC6, TIM_USE_MOTOR, 0, 0),   // S1   DMA2_ST2 (XXX was DMA1_ST4)
+    DEF_TIM(TIM8, CH1, PC6, TIM_USE_MOTOR, 0, 1),   // S1   DMA2_ST2 (XXX was DMA1_ST4)
```

Comment:
What is the purpose of this change? As far as I can tell this only moves this timer from one channel on DMA2 Stream 2 to another channel on the same stream, not doing anything to avoid conflict.

Reply:
Yeah @mikeller, it's true. It's only a dmaopt change, but an important one. Let me show you the current config and the new one:

Old:
```
Score: 221 m: 8, dmatt: 1 emc: 0 lmc: 0, ledmc 0, odmac: 2 adcc: 0 ledc: 0 tc: 1 tcc: 0 nt 3 mxt 2
   PA3,    PC6,    PC7,    PC8,    PC9,    PB1,    PA8,    PB8,    PA2,    PA0,    PA1,   PA15,   ADC
T09 C2, T08 C1, T08 C2, T08 C3, T08 C4, T03 C4, T01 C1, T04 C3, T05 C3, T05 C1, T05 C2, T02 C1, ADC1 
D0S0O0, D2S2O0, D2S3O1, D2S4O1, D2S7O0, D1S2O0, D2S6O0, D1S7O0, D1S0O0, D1S2O0, D1S4O0, D1S5O0, D2 S0 
Usable motors:  M1 M2 M3 M4 M5 M6 M7 M8
```

New:
```
Score: 21 m: 8, dmatt: 0 emc: 0 lmc: 0, ledmc 0, odmac: 2 adcc: 0 ledc: 0 tc: 1 tcc: 0 nt 3 mxt 2
   PA3,    PC6,    PC7,    PC8,    PC9,    PB1,    PA8,    PB8,    PA2,    PA0,    PA1,   PA15,   ADC
T09 C2, T08 C1, T08 C2, T08 C3, T08 C4, T03 C4, T01 C1, T04 C3, T05 C3, T05 C1, T05 C2, T02 C1, ADC1 
D0S0O0, D2S2O1, D2S3O1, D2S4O1, D2S7O0, D1S2O0, D2S6O0, D1S7O0, D1S0O0, D1S2O0, D1S4O0, D1S5O0, D2 S0 
Usable motors:  M1 M2 M3 M4 M5 M6 M7 M8
```

The main difference is the dmatt conflict reported in the current config. That means that a dma stream/channel which serves multiple timer channels is used and more than one of these channels is active in the config.

Looking at the dma options for timer 8 and f7:

```
    Timer, Channel, Dma, Stream, Channel, Dmaopt
    8,1,2,2,0,0,
    8,1,2,2,7,1,
    8,2,2,2,0,0,
    8,2,2,3,7,1,
    8,3,2,2,0,0,
    8,3,2,4,7,1,
    8,4,2,7,7,0,
```

You can see the problem: Dma 2, Stream 2, Channel 0 serves Timer 8, Channel 1-3. The config has all of these timer channels active, so this dmaopt cannot be used since you would otherwise get errors as the dma stream gets triggered for all of Timer8Ch1-3. Yet it is in the current config. Which means that non burst dshot does currently not work.

Reply:
Heh, I see. I wasn't aware of these constraints with respect to DMA channels, as they don't apply to DMA used for output - which is all we've used DMA for in Dshot before bidirectional Dshot came along.

Reply:
Actually, @mikeller, they do apply for output as well. Just not with dshot burst. If you switch the matekf722 to dshot_burst=off it will no longer spin all motors. But yeah learning all these constraints has been a painful part of the implementation of bidir. I'm still not sure I know all of them.

Reply:
Wait, I still don't quite understand in this case. What is the limitation that was violated here? Is it that DMA / channel / stream slots cannot be used for a timer if the slot specifies more than one timer, and more than one of them is used?

Re constraints, there are also the ones that are not / poorly documented, like that there is an upper limit on how many DMA operations can be used concurrently...

Reply:
> Is it that DMA / channel / stream slots cannot be used for a timer if the slot specifies more than one timer, and more than one of them is used?

Exactly.

Reply:
Good to know, thanks.